pkg/module/source/git.go

```diff
@@ -0,0 +1,123 @@
+package source
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/accessmethods/github"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/attrs/compatattr"
+	ocm "github.com/open-component-model/ocm/pkg/contexts/ocm/compdesc"
+	ocmv1 "github.com/open-component-model/ocm/pkg/contexts/ocm/compdesc/meta/v1"
+	"github.com/open-component-model/ocm/pkg/contexts/ocm/cpi"
+)
+
+const (
+	gitFolder = ".git"
+	refLabel  = "git.kyma-project.io/ref"
+)
+
+var (
+	errNotGit = errors.New("not a git repository")
+)
+
+type GitSource struct{}
+
+func NewGitSource() *GitSource {
+	return &GitSource{}
+}
+
+func (g GitSource) FetchSource(ctx cpi.Context, path, repo, version string) (*ocm.Source, error) {
+	ref, commit, err := g.getGitInfo(path)
+	if err != nil {
+		return nil, err
+	}
+
+	repo, err = g.determineRepositoryURL(repo)
+	if err != nil {
+		return nil, err
+	}
+
+	sourceType := "git"
+	if !compatattr.Get(ctx) {
+		sourceType = github.CONSUMER_TYPE
+	}
+
+	label, err := ocmv1.NewLabel(refLabel, ref, ocmv1.WithVersion(ocmVersion))
+	if err != nil {
+		return nil, err
+	}
+
+	access := github.New(repo, "", commit)
+
+	sourceMeta := ocm.SourceMeta{
+		Type: sourceType,
+		ElementMeta: ocm.ElementMeta{
+			Name:    ocmIdentityName,
+			Version: version,
+			Labels:  ocmv1.Labels{*label},
+		},
+	}
+
+	return &ocm.Source{
+		SourceMeta: sourceMeta,
+		Access:     access,
+	}, nil
+}
+
+func (g GitSource) determineRepositoryURL(repo string) (string, error) {
+	if repo == "" {
```

Comment:
In general, if you invert conditions like this you can work with early returns to reduce nesting and increase readability/maintainability.